homeassistant/components/speedtestdotnet/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Config flow for Transmission Bittorent Client."""
+import speedtest
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.core import callback
+
+from .const import (
+    CONF_MANUAL,
+    CONF_SERVER_ID,
+    DEFAULT_NAME,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+
+class SpeedTestFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle SpeedTest config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return SpeedTestOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="one_instance_allowed")
+
+        if user_input is None:
+            return self.async_show_form(step_id="user")
+
+        return self.async_create_entry(title=DEFAULT_NAME, data=user_input)
+
+    async def async_step_import(self, import_config):
+        """Import from SpeedTest config."""
+        import_config[CONF_SCAN_INTERVAL] = (
+            import_config[CONF_SCAN_INTERVAL].seconds / 60
+        )
+        if import_config[CONF_SCAN_INTERVAL] < 1:
+            import_config[CONF_SCAN_INTERVAL] = 1
+
+        return await self.async_step_user(user_input=import_config)
+
+
+class SpeedTestOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle SpeedTest options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        errors = {}
+        if user_input is not None:
+
+            try:
+                api = speedtest.Speedtest()
+                api.get_servers([user_input[CONF_SERVER_ID]])
+                return self.async_create_entry(title="", data=user_input)
+            except speedtest.NoMatchedServers:
+                errors[CONF_SERVER_ID] = "wrong_serverid"
+
+        options = {
+            vol.Optional(
+                CONF_SERVER_ID,
+                default=self.config_entry.options.get(CONF_SERVER_ID, ""),
+            ): int,
```

Comment:
How can an int accept an empty string as a default?

Reply:
This was what i wanted help with in my initial comment. Is it possible to save an int as blank because for the server_id, if no id is defined we can use the `get_best_server()`. But I don't know how we can save this option as blank. Can you please suggest.